components/proxy-service/internal/metadata/serviceapi/serviceapiservice.go

```diff
@@ -19,7 +19,13 @@ type API struct {
 // Credentials contains OAuth configuration.
 type Credentials struct {
 	// Oauth is OAuth configuration.
-	Oauth Oauth
+	Oauth *Oauth
+	Basic *Basic
+}
+
+type Basic struct {
```

Comment:
Please add a comment here (or remove the rest) to keep them consistent across this file.